flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/delegation/ParserImplTest.java

```diff
@@ -0,0 +1,177 @@
+import org.apache.flink.table.operations.command.SetOperation;
+import org.apache.flink.table.operations.command.SourceOperation;
+import org.apache.flink.table.planner.calcite.FlinkPlannerImpl;
+import org.apache.flink.table.planner.catalog.CatalogManagerCalciteSchema;
+import org.apache.flink.table.utils.CatalogManagerMocks;
+
+import org.hamcrest.Matcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.apache.calcite.jdbc.CalciteSchemaBuilder.asRootSchema;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+/** Test for {@link ParserImpl}. */
+public class ParserImplTest {
+
+    @Rule public ExpectedException thrown = ExpectedException.none();
+
+    private final boolean isStreamingMode = false;
+    private final TableConfig tableConfig = new TableConfig();
+    private final Catalog catalog = new GenericInMemoryCatalog("MockCatalog", "default");
+    private final CatalogManager catalogManager =
+            CatalogManagerMocks.preparedCatalogManager().defaultCatalog("builtin", catalog).build();
+    private final ModuleManager moduleManager = new ModuleManager();
+    private final FunctionCatalog functionCatalog =
+            new FunctionCatalog(tableConfig, catalogManager, moduleManager);
+    private final PlannerContext plannerContext =
+            new PlannerContext(
+                    tableConfig,
+                    functionCatalog,
+                    catalogManager,
+                    asRootSchema(new CatalogManagerCalciteSchema(catalogManager, isStreamingMode)),
+                    new ArrayList<>());
+
+    private final Supplier<FlinkPlannerImpl> plannerSupplier =
+            () ->
+                    plannerContext.createFlinkPlanner(
+                            catalogManager.getCurrentCatalog(),
+                            catalogManager.getCurrentDatabase());
+
+    private final Parser parser =
+            new ParserImpl(
+                    catalogManager,
+                    plannerSupplier,
+                    () -> plannerSupplier.get().parser(),
+                    t ->
+                            plannerContext.createSqlExprToRexConverter(
+                                    plannerContext.getTypeFactory().buildRelNodeRowType(t)));
+
+    @Test
+    public void testClearCommand() {
+        assertSimpleCommand("ClEaR", instanceOf(ClearOperation.class));
+    }
+
+    @Test
+    public void testHelpCommand() {
+        assertSimpleCommand("hELp", instanceOf(HelpOperation.class));
+    }
+
+    @Test
+    public void testQuitCommand() {
+        assertSimpleCommand("qUIt", instanceOf(QuitOperation.class));
+        assertSimpleCommand("Exit", instanceOf(QuitOperation.class));
```

Comment:
Would be better to refactor the tests into `Parameterized` `TestSpec`, see: `org.apache.flink.table.types.DataTypesTest`. The `TestSpec` can simply compare the parsed Operation with expected Operation using `asSummaryString`. 

That would be easier to add new command parse in the future.

Reply:
Besides, please also add tests for statemenets parsed by Calcite.